File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum=0;
        int maxi=Integer.MIN_VALUE;
        for(int i:nums){
            sum+=i;
            maxi=Math.max(sum,maxi);
            if(sum<0)
                sum=0;
            
        }
        return maxi;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxi=float('-inf')
        sumi=0
        for i in nums:
            sumi+=i
            maxi=max(maxi,sumi)
            if sumi<0:
                sumi=0
        return maxi
                